src/core/meshcop/dataset_manager.cpp

```diff
@@ -159,30 +160,44 @@ otError DatasetManager::Set(const Dataset &aDataset)
 
         if (mLocal.GetType() == Tlv::kActiveTimestamp)
         {
-            SuccessOrExit(error = aDataset.ApplyConfiguration(GetInstance()));
+            if (aDataset.Get(Tlv::kPendingTimestamp))
```

Comment:
If you want to judge the pending dataset from the Timestamp tlv, I think it may fail, you can check this:
https://github.com/openthread/openthread/blob/master/src/core/meshcop/dataset_manager.cpp#L524
before setting pending dataset to active dataset, the Tlv::kPendingTimestamp is removed.
Right?